frappe/model/base_document.py

```diff
@@ -225,12 +225,15 @@ def _init_child(self, value, key):
 		if not self.doctype:
 			return value
 
-		if not isinstance(value, BaseDocument):
-			value["doctype"] = self.get_table_field_doctype(key)
-			if not value["doctype"]:
-				raise AttributeError(key)
+		child_doctype = self.get_table_field_doctype(key)
+		if not child_doctype:
+			# probably just a list of values like __print_formats, don't init
+			# this should ideally be tested in set() above
+			return value
```

Comment:
This can alternatively be put inside the `if not isinstance(value, BaseDocument):` block below. Although, not sure why we want to update `parent`, `parenttype`, `parentfield`, `docstatus`, `idx` if it is not a child doc.